src/main/java/jenkins/plugins/office365connector/model/adaptivecard/AdaptiveCardAction.java

```diff
@@ -0,0 +1,56 @@
+package jenkins.plugins.office365connector.model.adaptivecard;
+
+import java.util.List;
+
+public class AdaptiveCardAction implements jenkins.plugins.office365connector.model.Action {
+
+    private String type = "Action.OpenUrl";
+
+    private String title;
+
+    private String url;
+
+    public AdaptiveCardAction(String title, String url) {
+        this.title = title;
+        this.url = url;
+    }
+
+    public String getType() {
+        return type;
+    }
+
+    public void setType(final String type) {
+        this.type = type;
+    }
+
+    public String getTitle() {
+        return title;
+    }
+
+    public void setTitle(final String title) {
+        this.title = title;
+    }
+
+    public String getUrl() {
+        return url;
+    }
+
+    public void setUrl(final String url) {
+        this.url = url;
+    }
+
+    @Override
+    public void setName(final String name) {
+        setTitle(name);
+    }
+
+    @Override
+    public void setTarget(final List<String> target) {
+        target.stream().findFirst().ifPresent(this::setUrl);
```

Comment:
applies only to first element, why is that ?

Reply:
Old `PotentialAction` works internally with list https://github.com/jenkinsci/office-365-connector-plugin/blob/master/src/main/java/jenkins/plugins/office365connector/model/PotentialAction.java#L34 i assume the `MessageCard` format expects lists. But the `AdaptiveCard`  action AFAIK doesn't. So that's why internally `url` (none-list) is used. As far as i can see, the whole code doesn't add more then one anyway. So a bigger refactoring could be, work internally with single attribute and only modify the `getTarget` in `MessageCard`  to return a list to fulfil the MessageCard JSON format.